### PR TITLE
Small changes regarding input file parsing

### PR DESCRIPTION
**Pull request by kmokstad**
Mostly to be aligned with the wiki documentation, which I updated earlier this summer.

### CODE REVIEW

File: src/SIM/SIMoutput.C [hunk 1]
```diff
@@ -393,9 +396,10 @@ bool SIMoutput::writeGlvBC (int& nBlock, int iStep) const
       continue; // nothing on this patch
 
     if (msgLevel > 1)
-      IFEM::cout <<"Writing boundary conditions for patch "<< i+1 << std::endl;
+      IFEM::cout <<"Writing boundary conditions for patch "
+                 << pch->idx+1 << std::endl;
```
Comment (akva2) [inline src/SIM/SIMoutput.C]: isnt' pch->idx 1 based?

File: src/SIM/SIMoutput.C [hunk 2]
```diff
@@ -761,7 +765,8 @@ bool SIMoutput::writeGlvS2 (const Vector& psol, int iStep, int& nBlock,
       continue; // skip empty patches
 
     if (msgLevel > 1)
-      IFEM::cout <<"Writing secondary solution for patch "<< i+1 << std::endl;
+      IFEM::cout <<"Writing secondary solution for patch "
+                 << myModel[i]->idx+1 << std::endl;
```
Comment (akva2) [inline src/SIM/SIMoutput.C]: same, i thought idx is 1 based.

File: src/SIM/SIMoutput.C [hunk 3]
```diff
@@ -1033,13 +1037,13 @@ bool SIMoutput::writeGlvF (const RealFunc& f, const char* fname,
...
+  for (const ASMbase* pch : myModel)
   {
-    if (myModel[i]->empty()) continue; // skip empty patches
+    if (pch->empty()) continue; // skip empty patches
 
     if (msgLevel > 1)
-      IFEM::cout <<"Writing function "<< fname
-                 <<" for patch "<< i+1 << std::endl;
+      IFEM::cout <<"Writing function '"<< fname
+                 <<"' for patch "<< pch->idx+1 << std::endl;
```
Comment (akva2) [inline src/SIM/SIMoutput.C]: same.

File: src/SIM/SIMoutput.C [hunk 4]
```diff
@@ -1238,8 +1242,8 @@ bool SIMoutput::writeGlvE (const Vector& vec, int iStep, int& nBlock,
     if (pch->empty()) continue; // skip empty patches
 
     if (msgLevel > 1)
-      IFEM::cout <<"Writing element field "<< name
-                 <<" for patch "<< pch->idx+1 << std::endl;
+      IFEM::cout <<"Writing element field '"<< name
+                 <<"' for patch "<< pch->idx+1 << std::endl;
```
Comment (akva2) [inline src/SIM/SIMoutput.C]: same

File: src/SIM/SIMoutput.C [hunk 5]
```diff
@@ -1412,7 +1415,7 @@ bool SIMoutput::dumpSolution (const Vector& psol, utl::LogStream& os) const
     if (myModel[i]->empty()) continue; // skip empty patches
 
     if (myModel.size() > 1)
-      os <<"\n# Patch: "<< i+1;
+      os <<"\n# Patch: "<< myModel[i]->idx+1;
```
Comment (akva2) [inline src/SIM/SIMoutput.C]: same.

File: src/SIM/SIMoutput.C (hunk 1 above)
Comment (kmokstad) [reply src/SIM/SIMoutput.C]: Obviously, I thought here it was zero-based. But now you made me unsure, so I check again:

It is assigned in the ASMs[123]D::readPatch() methods, e.g.
ASMs3D.C:718       `pch->idx = myModel.size();`

This happens _before_ that patch is added to myModel (which is done by the caller). So then it must be zero-based as the myModel is empty when reading the first patch. Also all other uses of pch->idx indicates it is zero-based.

Comment (akva2) [reply src/SIM/SIMoutput.C]: aha, i missed that it happened before. then it's obviously fine.

**Pull request merged**